substrate/frame/child-bounties/src/lib.rs

```diff
@@ -905,8 +935,11 @@ impl<T: Config> pallet_bounties::ChildBountyManager<BalanceOf<T>> for Pallet<T>
 	fn children_curator_fees(bounty_id: pallet_bounties::BountyIndex) -> BalanceOf<T> {
 		// This is asked for when the parent bounty is being claimed. No use of
 		// keeping it in state after that. Hence removing.
-		let children_fee_total = ChildrenCuratorFees::<T>::get(bounty_id);
-		ChildrenCuratorFees::<T>::remove(bounty_id);
+		let mut children_fee_total = 0u32.into();
+		for child_fee in ChildrenCuratorFees::<T>::iter_prefix_values(bounty_id) {
+			children_fee_total += child_fee;
+		}
+		ChildrenCuratorFees::<T>::drain_prefix(bounty_id);
```

Comment:
lets update the doc for the trait and mentioned that this function not only returns the fee, it also removes it from the source.